tools/rc/usResourceCompiler.cpp

```diff
@@ -532,107 +531,74 @@ int main(int argc, char** argv)
   // ---------------------------------------------------------------------------------
 
   // Create a new zip archive which will be copied to zipfile later
-  dbg_print("Creating zip archive\n");
-  remove(zipFile);
+  std::clog << "Creating zip archive " << outfile.c_str() << std::endl;
+  remove(outfile.c_str());
 
-  if (!mz_zip_writer_init_file(&writeArchive, zipFile, 0))
+  if (!mz_zip_writer_init_file(&writeArchive, outfile.c_str(), 0))
   {
     exit_printf(&writeArchive, "Internal error, could not init new zip archive\n");
   }
-  dbg_print("Initialized zip archive\n");
+  std::clog << "Initialized zip archive" << std::endl;
 
-  // Add current files to the zip archive
-  zipArgIndices = malloc_or_abort(argc * sizeof *zipArgIndices);
-  while(++argIndex < argc)
+  // Add resource files to the zip archive
+  for (option::Option* opt = options[RESFILE]; opt; opt = opt->next())
   {
-    const char* fileName = argv[argIndex];
-    const size_t fileNameLength = strlen(fileName);
-
-    // determine the argument type
-    if (strcmp(argv[argIndex], "-a") == 0)
-    {
-      mergeFlag = 0;
-      continue;
+    if (!bundleName.size()) {
+      std::cerr << "No bundle name specified ... cannot add resource files to zip archive" << std::endl;
+      return EXIT_FAILURE;
     }
-    else if (strcmp(argv[argIndex], "-m") == 0)
-    {
-      mergeFlag = 1;
-      continue;
-    }
-
-    if (mergeFlag)
-    {
-      // check if the current file is a valid zip archive
-      memset(&currFileArchive, 0, sizeof(currFileArchive));
-      if (mz_zip_reader_init_file(&currFileArchive, fileName, 0))
-      {
-        dbg_print("Input is a valid zip archive: %s\n", fileName);
-        zipArgIndices[numZipArgs++] = argIndex;
-        mz_zip_reader_end(&currFileArchive);
-      }
-      // silently ignore files which are not zip archives
-      continue;
-    }
-
-    if (fileNameLength + 1 + bundleNameLength > MZ_ZIP_MAX_ARCHIVE_FILENAME_SIZE - 1)
-    {
-      exit_printf(&writeArchive, "Resource filename too long: %s\n", bundleName);
-    }
-    US_STRCPY(archiveName, sizeof archiveName, bundleName);
-    archiveName[bundleNameLength] = '/';
-    US_STRCPY(archiveName + bundleNameLength + 1, (sizeof archiveName) - (bundleNameLength + 1), fileName);
-
-    printf("  adding: %s\n", archiveName);
-
+    std::string resArchiveName = opt->arg;
+    resArchiveName.insert(0, "/");
+    resArchiveName.insert(0, bundleName);
+    
+    std::clog << "  adding: " << resArchiveName.c_str() << std::endl;
```

Comment:
The .c_str() should be unnecessary here as the stream should be able to handle a std::string.